glib/src/object.rs

```diff
@@ -1525,9 +1525,78 @@ impl<'a, O: IsA<Object> + IsClass> ObjectBuilder<'a, O> {
     }
 
     // rustdoc-stripper-ignore-next
-    /// Set property `name` to the given value `value`.
+    /// Build the object with the provided properties.
+    ///
+    /// # Panics
+    ///
+    /// This panics if the object is not instantiable, doesn't have all the given properties or
+    /// property values of the wrong type are provided.
+    #[track_caller]
+    #[inline]
+    pub fn build(mut self) -> O {
+        let object = Object::with_mut_values(self.type_, &mut self.properties);
+        unsafe { object.unsafe_cast::<O>() }
+    }
+}
+
+pub trait ObjectBuilderPropertySetter<'a>: Sized {
+    // rustdoc-stripper-ignore-next
+    /// Sets property `name` to the given value `value`.
+    ///
+    /// Overrides any default or previously defined value.
+    fn property(self, name: &'a str, value: impl Into<Value>) -> Self;
+
+    // rustdoc-stripper-ignore-next
+    /// Sets property `name` to the given inner value if `value` is `Some`.
+    ///
+    /// This has no effect if the value is `None`, i.e. default or previous value is kept.
+    #[inline]
+    fn property_if_some(self, name: &'a str, value: Option<impl Into<Value>>) -> Self {
+        if let Some(value) = value {
+            ObjectBuilderPropertySetter::property(self, name, value)
+        } else {
+            self
+        }
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Sets property `name` using the given `ValueType` `V` built from `iter`'s the `Item`s.
+    ///
+    /// Overrides any default or previously defined value.
     #[inline]
-    pub fn property(self, name: &'a str, value: impl Into<Value>) -> Self {
+    fn property_from_iter<V: ValueType + Into<Value> + FromIterator<SendValue>>(
+        self,
+        name: &'a str,
+        iter: impl IntoIterator<Item = impl Into<SendValue>>,
```

Comment:
```suggestion
        iter: impl IntoIterator<Item = impl Into<Value>>,
```

just like for the other property methods

Reply:
See discussion in https://github.com/gtk-rs/gtk-rs-core/pull/1377#pullrequestreview-2030319735 & https://github.com/gtk-rs/gtk-rs-core/pull/1377#pullrequestreview-2030316501